src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
+
+    this.logger.debug(
+      `Request Method:[${req.method}] || Request URI:${
+        req.baseUrl
+      } || Request Body:[${JSON.stringify(
+        req.body,
+      )}] || Request Query Params:[${JSON.stringify(req.query)}] Request IP:[${
+        req.ip
+      }] || User Details: [${JSON.stringify(userDetail)}]`,
+    );
```

Comment:
PullProofAI

The logging logic can be improved by breaking down the message into smaller parts for better readability and maintainability.
```diff
-     this.logger.debug(
-       `Request Method:[${req.method}] || Request URI:${
-         req.baseUrl
-       } || Request Body:[${JSON.stringify(
-         req.body,
-       )}] || Request Query Params:[${JSON.stringify(req.query)}] Request IP:[${
-         req.ip
-       }] || User Details: [${JSON.stringify(userDetail)}]`,
+     const logMessage = `Request Method: [${req.method}] || Request URI: ${req.baseUrl}`;
+     const requestBody = `Request Body: ${JSON.stringify(req.body)}`;
+     const queryParams = `Request Query Params: ${JSON.stringify(req.query)}`;
+     const requestIP = `Request IP: ${req.ip}`;
+     const userDetails = `User Details: ${JSON.stringify(userDetail)}`;
+     
+     this.logger.debug(`${logMessage} || ${requestBody} || ${queryParams} || ${requestIP} || ${userDetails}`);
```



<!-- This is an auto-generated comment by PullProofAI -->